src/Architectures.jl

```diff
@@ -27,21 +28,49 @@ struct CPU <: AbstractArchitecture end
 """
     GPU <: AbstractArchitecture
 
-Run Oceananigans on a single NVIDIA CUDA GPU.
+Run Oceananigans on a single GPU `device`.
 """
-struct GPU <: AbstractArchitecture end
+struct GPU{D} <: AbstractArchitecture 
+    device :: D
+end
+
+function GPU()
+    if CUDA.has_cuda()
+        return GPU(CUDA.CUDABackend(; always_inline = true))
+    elseif has_metal_device()
+        return GPU(Metal.MetalBackend())
+    else
+        error("No compatible GPU detected")
+    end
+end
```

Comment:
Make the device selectable:

```suggestion
function GPU(device=nothing)
    if isnothing(device) # try to find one
        device = if CUDA.has_cuda()
            CUDA.CUDABackend(; always_inline = true)
        elseif has_metal_device()
            Metal.MetalBackend()
        else
            throw(ArgumentError("Cannot construct GPU architecture because no compatible GPU was found!"))
        end
    end
        
    return device
end
```

It also needs a docstring.